### DISCUSSION TITLE
Support loading graph from HuaWei Cloud OBS？

### DISCUSSION THREAD

**Discussion by JackyYangPassion** [alibaba/GraphScope#3314] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
i want to load graph from datawarhouse where the ORC file Store in the obs

**Describe the solution you'd like**

```
graph = (
        graph.add_vertices(Loader("obs:///data1/DataWareHouse/graph/dt=2023-10-23/000000_0",filetype="ORC")
                           ,label="user"
                           ,vid_field="user_id"
                          )
    )
```

Is this syntax currently supported or are there other ways to write ways to write it?

**Comment by siyuan0322** [comment]
Does OBS support be read from a s3 sdk (boto3)? If it is, just change prefix to 's3' and give it a try.

**Comment by siyuan0322** [comment]
And i support the file should end in .orc. we currently doesn't support assigning a filetype.

**Comment by JackyYangPassion** [comment]
> And i support the file should end in .orc. we currently doesn't support assigning a filetype.

GraphScope==0.24.0 Support filetype="ORC".

Does it mean that future new versions will not support the specified parameter filetype="ORC"?

**Comment by siyuan0322** [comment]
Oh, sorry, I made a mistake. It's supported.

**Comment by JackyYangPassion** [comment]
> Does OBS support be read from a s3 sdk (boto3)? If it is, just change prefix to 's3' and give it a try.

Thanks for your reply.

Loader("s3://datafiles/group.e", key='access-id', secret='secret-access-key', client_kwargs={'region_name': 'us-east-1'})

Could you guide me on how to set up the Endpoint?

**Comment by siyuan0322** [comment, marked as answer]
Unfortunately I'm not familiar with OBS...
After some searching for obs + s3, obs + boto3, and the sdk provided by huawei cloud, I didn't find any resources..

If OBS is a must, you may consider implement a OBS driver follow apis of fsspec, here's a example of [OSS](https://github.com/fsspec/ossfs)
